src/Microsoft.AspNetCore.Razor.Evolution/DefaultRazorIRLoweringPhase.cs

```diff
@@ -126,7 +126,30 @@ public override void VisitStartTemplateBlock(TemplateBlockChunkGenerator chunkGe
 
             public override void VisitEndTemplateBlock(TemplateBlockChunkGenerator chunkGenerator, Block block)
             {
-                Builder.Pop();
+                var templateNode = Builder.Pop();
+                if (templateNode.Children.Count > 0)
```

Comment:
Talked offline about this. We still want to ensure what the user wrote is flowed through the IR.